server/api/resolvers/donorHomepageResolvers.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    Banner,
+    DonorHomepage,
+    DonorHomepageInterface,
+    Map,
+    Statistic,
+    StatisticMeasurement,
+    Testimonial,
+    TestimonialCarousel
+} from "../../database/models/donorHomepageModel";
+
+const getDonorHomepageData = async <Type>(property: string): Promise<Type> => {
+    return DonorHomepage.find()
+        .exec()
+        .then((donorHomepageData) => {
+            return donorHomepageData[0][property];
+        });
+};
+
+const donorHomepageQueryResolvers = {
+    donorHomepageBanner: async (): Promise<Banner> => {
+        return getDonorHomepageData<Banner>("banner");
+    },
+    donorHomepageTestimonialCarousel: async (): Promise<TestimonialCarousel> => {
+        return getDonorHomepageData<TestimonialCarousel>("testimonialCarousel");
+    },
+    donorHomepageMap: async (): Promise<Map> => {
+        return getDonorHomepageData<Map>("map");
+    },
+    donorHomepageStatistics: async (): Promise<Array<Statistic>> => {
+        return getDonorHomepageData<Array<Statistic>>("statistics");
+    }
+};
+
+const donorHomepageMutationResolvers = {
+    updateDonorHomepage: async (
+        _,
+        { mapTestimonials, carouselTestimonials, statMeasurements },
+        { authenticateUser }
+    ): Promise<DonorHomepageInterface> => {
+        return authenticateUser().then(async () => {
+            const map = await donorHomepageQueryResolvers.donorHomepageMap();
+            map.testimonials = mapTestimonials;
+
+            const statistics = await donorHomepageQueryResolvers.donorHomepageStatistics();
+            statistics.forEach((statistic) => {
+                const type = statistic.type.toString();
+                if (statMeasurements[type] !== "" || statMeasurements[type] === null) {
```

Comment:
this condition should be `statMeasurements[type] !== "" && statMeasurements[type] != null` bc we want the measurement to be a valid string in order to perform the update, and the `!=` condition checks for both `statMeasurement[type] === null` and `statMeasurement === undefined`

Reply:
oh right omg LOL mb that makes sense thank you!!